tests/testthat/test-paths.R

```diff
@@ -0,0 +1,64 @@
+
+context("non-trivial paths")
+
+test_that("folders with potentially problematic characters", {
+
+  skip_on_cran()
+
+  tmp <- tempfile()
+  on.exit(tryCatch(pkgload::unload("foo"), error = identity), add = TRUE)
+  on.exit(unlink(tmp, recursive = TRUE), add = TRUE)
+  on.exit(environment(need_internal_tar)$internal <- NULL, add = TRUE)
+
+  pkg <- binary_test_package("foo_0.0.0.9000")
+
+  folders <- c(
+    "s p a c e s",
+    enc2native("\u00fa\u00e1\u00f6\u0151\u00e9"),
+    "s' p' a' c' e' s'"
+  )
+
+  skipped <- 0
+
+  for (f in folders) {
+    error <- FALSE
+    tryCatch(
+      {
+        unloadNamespace("foo")
+        unlink(tmp, recursive = TRUE)
+        dir.create(tmp)
+        dir.create(file.path(tmp, f))
+        libdir <- dir(tmp)
+        libpath <- file.path(tmp, libdir)
+      },
+      warning = function(e) error <<- TRUE,
+      error = function(e) error <<- TRUE
+    )
+    if (error) { skipped <- skipped + 1; next }
+
+    ## Reset this
+    environment(need_internal_tar)$internal <- NULL
+
+    expect_error_free(install_binary(pkg, lib = libpath, quiet = TRUE))
+    expect_error_free(library("foo", lib.loc = libpath))
+    expect_equal(foo::foo(), NULL)
+    tryCatch(pkgload::unload("foo"), error = identity)
```

Comment:
Do you actually need `pkgload::unload()`? just `detach(unload = TRUE)` is not enough?

Reply:
It is not enough because the shared lib is locked on windows

Reply:
Even if you explicitly call `library.dynam.unload()` in `.onUnload()`?

Reply:
Maybe we just need to add that in https://github.com/r-lib/pkginstall/blob/master/tests/testthat/foo/R/foo.R then?

Reply:
I can give it a try.

Reply:
Yep, that seems to work as well. There was actually a `pkgload::unload()` call in `helper.R` already, I replaced that as well.